pkg/handlers/handlers_test.go

```diff
@@ -684,3 +682,144 @@ func setupTempGitRepoCommon(t *testing.T, fileName string, fileSize int, isUnsup
 
 	return tempDir
 }
+
+// errorOnceReader is a custom io.Reader that simulates a read error only once.
+// On the first read, it returns an error. Subsequent reads return the remaining data normally.
+// This is used to simulate an error during mime type detection in newFileReader.
+type errorOnceReader struct {
```

Comment:
@mcastorina I feel like I did something like this before and you mentioned there was a pkg to do this, or maybe i'm thinking of something else?

Reply:
I'll usually look in [testing/iotest](https://pkg.go.dev/testing/iotest) for stuff like that, but I don't see any similar to this functionality.